### DISCUSSION TITLE
Shadow return codes?

### DISCUSSION THREAD

**Discussion by timwalsh300** [shadow/shadow#2654] [Q&A] [answered]
Is there a list of Shadow return codes and their meanings available somewhere? Working with tornettools, I've had simulations terminate with various return codes from Shadow, maybe 1, 4, and 7...

**Comment by sporksmith** [comment, marked as answer]
The short answer is no, sorry, there isn't :)

I didn't know we were actually using multiple return codes. shadow returns a non-zero exit code when a managed program exits with a non-zero exit code; maybe we're actually propagating the specific code.

In any case, I'd check the shadow log. There ought to be an `ERROR` level log near the end to tell you why Shadow exited with an error.
